Synview-backend/routes/user_routes.ts

```diff
@@ -0,0 +1,170 @@
+import { withAccelerate, PrismaClient, Router, oakCors } from "../deps.ts";
+import { hash, bycryptVerify } from "../deps.ts";
+import {
+  createToken,
+  getPayload,
+  getPayloadFromToken,
+} from "../middleware/auth_middleware.ts";
+import token from "../middleware/token.ts";
+import { Session } from "../deps.ts";
+
+import AuthMiddleware from "../middleware/auth_middleware.ts";
+import { z } from "zod";
+type AppState = {
+  session: Session;
+};
+const router = new Router<AppState>();
+const prisma = new PrismaClient().$extends(withAccelerate());
+
+export const EmailRegisterRequestSchema = z.object({
+  ["username"]: z.string(),
+  ["email"]: z.string().email(),
+  ["password"]: z.string(),
+});
+export const EmailLoginRequestSchema = z.object({
+  ["email"]: z.string().email(),
+  ["password"]: z.string(),
+});
+
+///
+
+// router.use(
+//   oakCors({
+//     origin: "*",
+//     methods: ["POST", "PUT", "DELETE", "GET"],
+//     allowedHeaders: ["Content-type", "Authorization"],
+//   })
+// );
+// unprotectedRouter.use(
+//   oakCors({
+//     origin: "*",
+//     methods: ["POST", "PUT", "DELETE", "GET"],
+//     allowedHeaders: ["Content-type", "Authorization"],
+//   })
+// );
+
+router
+  .post("/register", async (context) => {
+    const body = await context.request.body.json();
+    const parsedBody = EmailRegisterRequestSchema.parse(body);
+
+    const { username, email, password } = parsedBody;
```

Comment:
```suggestion
    const { username, email, password } = EmailRegisterRequestSchema.parse(body);
```
Generally avoid intermediate steps unless they contain a lot of logic or you need the intermediate step.